### DISCUSSION TITLE
NEMO API access

### DISCUSSION THREAD

**Discussion by Hydrozyk** [usnistgov/NEMO#261] [Q&A]
Getting an error when accessing https://myhost.com/api Any insights on this would be greatly appreciated!

```
[19/Nov/2024 11:21:48] django.request WARNING Bad Request: /api/
[19/Nov/2024 11:21:52] django.request WARNING Bad Request: /api/
[19/Nov/2024 11:21:58] django.request WARNING Bad Request: /api
```

In settings.py I have:

```
REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': ('NEMO.permissions.BillingAPI',),
    'DEFAULT_FILTER_BACKENDS': ('django_filters.rest_framework.DjangoFilterBackend',),
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
    'DEFAULT_AUTHENTICATION_CLASSES': ('rest_framework.authentication.TokenAuthentication',
                                       'rest_framework.authentication.SessionAuthentication',),
    'DEFAULT_PERMISSION_CLASSES': ('rest_framework.permissions.IsAuthenticated',),
    "PAGE_SIZE": 1000,
}
```

```
INSTALLED_APPS = [
    ...
    'rest_framework',
    'rest_framework.authtoken',
    ...
]
```

For debugging purposes I did:

`ALLOWED_HOSTS = ['*']`

`CSRF_TRUSTED_ORIGINS = ['https://myhost.com']
`
Same error.

**Comment by rptmat57** [comment]
try with a slash at the end `https://myhost.com/api/`, just in case.

also, remove one of the 2 permission classes in your settings and try with this one:

```'DEFAULT_PERMISSION_CLASSES': ('NEMO.permissions.DjangoModelPermissions',),```

and check nemo's logs, there should be more information there about the exact error
